lib/FusionInventory/Agent/Tools/Linux.pm

```diff
@@ -385,54 +385,69 @@ sub _readLinkFromSysFs {
 sub getInfoFromSmartctl {
     my (%params) = @_;
 
-    my $handle = getFileHandle(
-        %params,
-        command => $params{device} ? "smartctl -i $params{device}" : undef,
-    );
-    return unless $handle;
+    $params{command} = $params{device} ? "smartctl -i $params{device} " : undef;
+    $params{command} .= $params{extra} if (defined $params{command} and defined $params{extra});
```

Comment:
```suggestion
    my @lines = getAllLines(
        %params,
        command => $params{device} ? "smartctl -i $params{device} ".($params{extra}//"") : undef,
    );
    return unless @lines;
```

Reply:
I would suggest to getAllLines() asap to return quickly when the command is not supported by the system. I discovered recently [the `//` operator](https://perldoc.perl.org/perlop#Logical-Defined-Or), you can also use it to make code lighter as in my code suggestion.

Reply:
I've known about `//` for a long time, but never used it much for some reason :)

Reply:
Updated the code.